cpp/arcticdb/column_store/column.hpp

```diff
@@ -646,6 +646,72 @@ class Column {
         return *res;
     }
 
+    template<typename input_tdt>
```

Comment:
I'm also wondering if Column has begin/end functions can't we use std::for_each instead of having our own version?

Reply:
It isn't clear in this PR, but I'm going to add some generalisations for sparse data that will be kept cleaner by hiding this detail inside the `Column` static methods.
In addition, hiding the iterators from the user makes it obvious that operations are happening over the whole column. We don't want users sub-selecting ranges inside columns, as this will not be efficient in general.